classes/Tools.php

```
@@ -725,43 +730,102 @@ public static function getCldr(Context $context = null, $language_code = null)
     }
 
     /**
-     * Return price with currency sign for a given product.
+     * Return price with currency sign for a given product
      *
-     * @param float $price Product price
-     * @param object|array $currency Current currency (object, id_currency, NULL => context currency)
+     * @deprecated Since 1.7.5.0. Please use Locale::formatPrice() instead
+     * @see PrestaShop\PrestaShop\Core\Localization\Locale
+     *
+     * @param float $price
+     *  Product price
+     *
+     * @param object|array $currency
+     *  Current currency (object, id_currency, NULL => context currency)
+     *
+     * @param bool $no_utf8
+     *  Not used anymore
      *
-     * @return string Price correctly formated (sign, decimal separator...)
-     *                if you modify this function, don't forget to modify the Javascript function formatCurrency (in tools.js)
+     * @param Context|null $context
+     *
+     * @return string Price correctly formatted (sign, decimal separator...)
+     * if you modify this function, don't forget to modify the Javascript function formatCurrency (in tools.js)
+     *
+     * @throws LocalizationException
      */
     public static function displayPrice($price, $currency = null, $no_utf8 = false, Context $context = null)
     {
+        @trigger_error(
```

Why adding `@`? We can let everyone editing `error_reporting` no?

---

for a deprecation I think it is safer to silently trigger the error

---

Still don't understand why we trigger an error if we hide it. 
Not sure people are watching logs, so if we hide we can remove and add a `@deprecated` annotation.

---

well to me the purpose of these kind of "silent errors" is to have logs which allow you to know what you need to migrate, is shows you error logs about what you are actually using
you also get notices in your tests which is a simple way to find what you forgot to migrate
I think this is what Symfony uses to manage their deprecated feature, it also allows to have deprecation notices in the debug bar
https://symfony.com/doc/current/contributing/code/conventions.html#deprecations